supervisor.go

```diff
@@ -0,0 +1,123 @@
+package streams
+
+// Supervisor represents a commit supervisor.
+//
+// The Supervisor performs a commit in a concurrently-safe manner.
+// There can only ever be 1 ongoing commit at any given time.
+type Supervisor interface {
+	// Perform a global commit sequence.
+	//
+	// If triggered by a Node, that node should be passed as an argument to Commit.
+	Commit(Processor) error
+}
+
+type sourceMetadata map[Source]Metadata
+
+// Merge merges metadata from metaitems.
+//
+// TODO: Incorporate merge strategies.
+func (m sourceMetadata) Merge(items []Metaitem) sourceMetadata {
+	for _, item := range items {
+		m[item.Source] = item.Metadata.Merge(m[item.Source])
+	}
+	return m
+}
+
+// Compile-type interface check.
+var _ Supervisor = (*supervisor)(nil)
```

Comment:
This is covered by the constructor.

Reply:
True. Then there are other places where we can remove those checks, I think ;)

Reply:
If you can find them, that would be awesome.